### DISCUSSION TITLE
Reading Fluent mesh warnings and erros

### DISCUSSION THREAD

**Discussion by giannoko** [pyvista/pyvista#3532] [Q&A] [answered]
Hi,

I am reading a mesh exported from ANSYS Fluent in .msh format with pyvista.read("mesh.msh"). I am getting a lot of warnings like these
```
Warning: Unknown index 4. Skipping.
Warning: Unknown index 11. Skipping.
Warning: Unknown index 11. Skipping.
Warning: Unknown index 11. Skipping.
Warning: Unknown index 316. Skipping.
Warning: Zone specification not supported yet. Skipping.
Warning: Zone specification not supported yet. Skipping.
Warning: Zone specification not supported yet. Skipping.
Warning: Zone specification not supported yet. Skipping.
Warning: Zone specification not supported yet. Skipping.
Warning: Zone specification not supported yet. Skipping.
Warning: Zone specification not supported yet. Skipping.
Warning: Unknown index 38. Skipping.
Warning: Unknown index 60. Skipping.
Warning: Unknown index 73. Skipping.
```
Although I can display the mesh, when I print the cell quality based on "skew", I got values of -1. According to documentation this means that _"Cell types not supported by this filter or undefined quality of supported cell types will have an entry of -1."_ The elements are tetrahedrons. I attach the mesh file in .txt format because I am not able to upload a .msh file. How can I compute and display the cell quality based on skewness for this mesh?

Thank you
[mesh.txt](https://github.com/pyvista/pyvista/files/9902530/mesh.txt)

**Comment by MatthewFlamm** [comment, marked as answer]
This is a mesh format that is not directly supported by PyVista.  See https://docs.pyvista.org/api/readers/_autosummary/pyvista.get_reader.html#pyvista.get_reader. If the mesh format isn't directly supported, it is attempted to read in by meshio https://github.com/nschloe/meshio.  So the warnings you see and the behavior is coming from meshio.  However, Im not sure whether this is a bug/missing feature in meshio or in the translation to the pyvista representation.  Maybe others would know.

I don't immediately see a vtk reader for this format, so I don't think pyvista can directly support this format.  Since you mention exporting from Fluent, you can also try to load in using a `.cas` file, for which there is existing support.

**Reply by giannoko** [reply]
Hi @MatthewFlamm. This is a meshio problem since I got the same warnings using the meshio reader out of PyVista. Although I translated the .msh file to .vtk file through meshio, I do not see any warning with PyVista reader but I still cannot plot the skewness propecrly (still values of -1).

**Reply by MatthewFlamm** [reply]
This sounds like a meshio problem to me unless the data is correct in meshio but gets corrupted when translating to pyvista.  If it is indeed a meshio problem, it would be best to report there.

**Reply by giannoko** [reply]
Thank you @MatthewFlamm I will report this to meshio.
